### PR TITLE
Improve local development and deployment experience

### PR DESCRIPTION
**Pull request by nonword**
Strives to improve local development and deployment experience by:
 - Clarify what this app does
 - Document how to start up the app locally and verify it's working (including limitations) using both docker-compose and sam
 - Update deployment concerns:
   - Update `node-lambda` to support Node14
   - Remove `dev` deployment path since we don't use it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,13 +5,10 @@
   "main": "index.js",
   "scripts": {
     "pretest": "phplint src/*.php src/**/*.php src/**/**/*.php",
-    "manual-deploy-development": "./node_modules/.bin/node-lambda deploy -e development -f config/var_development.env -S config/event_sources_development.json -b subnet-f4fe56af -g sg-1d544067 -P nypl-sandbox -o arn:aws:iam::224280085904:role/lambda_basic_execution --layers arn:aws:lambda:us-east-1:946183545209:layer:lib-for-node10-wrapped-php7:1",
     "manual-deploy-qa": "./node_modules/.bin/node-lambda deploy -e qa -f config/var_qa.env -S config/event_sources_qa.json -b subnet-f35de0a9,subnet-21a3b244 -g sg-aa74f1db -o arn:aws:iam::946183545209:role/lambda-full-access -P nypl-digital-dev --layers arn:aws:lambda:us-east-1:946183545209:layer:lib-for-node10-wrapped-php7:1",
```
Comment (charmingduchess) [inline package.json:8]: whats the difference between manual and not manual deploy?

Comment (nonword) [reply package.json:8]: Manual deploy uses a named profile (not available to travis). Not-manual deploy uses a key and secret (available to travis).

File: README.md [hunk 2]
```diff
@@ -5,200 +5,168 @@
...
-## Configuration
+Start local test postgres db:
 
-Various files are used to configure and deploy the Lambda.
+```
+docker-compose up -d db
+```
 
-### .env
+Edit `sam.local.yml` and change the `DB_CONNECT_STRING` to use your current public IP in place of "YOURIP". In OSX, your public IP is available via `ifconfig -u | grep 'inet ' | grep -v 127.0.0.1 | cut -d\  -f2 | head -1`.
```
Comment (danamansana) [inline README.md]: The `DB_CONNECT_STRING` in `sam.local.yml` does not seem to contain "YOURIP", can we clarify this?

**Review by danamansana — CHANGES_REQUESTED** [review]
This is great! Tests are working and I can get the server running no problem. But I think there is some mismatch between the README and the config regarding running SAM

File: README.md (hunk 2 above)
Comment (nonword) [reply README.md]: Good catch. Fixed

**Review by danamansana — APPROVED** [review]

**Pull request merged**